application/src/main/java/net/corda/v5/application/interop/evm/EvmService.java

```diff
@@ -0,0 +1,201 @@
+package net.corda.v5.application.interop.evm;
+
+import net.corda.v5.base.annotations.DoNotImplement;
+import net.corda.v5.base.annotations.Suspendable;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+
+/**
+ * Responsible for interactions with an Ethereum service
+ */
+@DoNotImplement
+public interface EvmService {
+
+    /**
+     * Sends a <a href="https://www.quicknode.com/docs/ethereum/eth_call">eth_call</a> query to the EVM.
+     *
+     * @param functionName    the function name on the EVM smart contract to call
+     * @param contractAddress the address of the smart contract
+     * @param parameters      any parameters (in order listed by the call) for the contract
+     * @return the returned value of the contract query
+     */
+    @Suspendable
+    @NotNull
+    Object call(@NotNull String contract,
```

Comment:
what is `contract` here?

Reply:
Ah, I didn't put that in the docs.  It should maybe be `contractName`.  It's basically how we know what the function we're going to encode is.  Ideally it would be something like

```
myContract.itsFunction()
```

but since we don't have autogeneration (yet) it's more like

```
evmService.call("myContract", "itsFunction)
```

I'll change the parameter to `contractName` and add the javadoc

Reply:
@fowlerrr I see, but is contractName a required param at all? A contract is identified by its deploymnt address, and the call's "entry point" by the hash of the function name and parameter types only.

Reply:
It's required by _us_ to know what function we're encoding.  Otherwise we can't properly encode its parameters.  Basically we'll use those first two parameters to find the matching `functionName` in `contractName`s ABI.  Then use that information from the ABI to encode the parameters correctly for sending to the external worker.

Reply:
But then you would have to rewrite all of the contracts that might exist.  The point here is to allow for a more generic approach.  If someone wanted to add a contract they'd just provide the ABI and that would be enough.

It's not ideal but it reduces the burden on them from having to write a Java version (especially since we can't use Web3j to generate the contract for us).

Eventually we'd prefer to have our own autogeneration, but that'll take more time than we have right now.